### DISCUSSION TITLE
How to properly do profiling?

### DISCUSSION THREAD

**Discussion by ngocphamm** [wbthomason/packer.nvim#1185] [Q&A] [answered]
I have been trying to profile my neovim load time with the built-in profiling feature of packer.

I run `nvim`, run `:PackerCompile profile=true`, restart `nvim`, run `:PackerProfile` but this is all I see. I'm fairly sure it doesn't show all my plugins.

![CleanShot 2023-01-03 at 22 48 22](https://user-images.githubusercontent.com/395123/210481055-cf7a7c00-16e6-4960-b8f5-7527df178c0f.png)

This is my `init.lua`, based off [kickstart.nvim](https://github.com/nvim-lua/kickstart.nvim)

https://gist.github.com/ngocphamm/151a0ab737eb93418b62ff3802de9b5c

**Comment by wbthomason** [comment, marked as answer]
Nope, that's actually accurate - `packer`'s profiling is specifically for the cost of `packer_compiled.lua`, where all of `packer`'s lazy-loading logic is handled. Since you don't have any lazy-loading, and only use a sequenced load (`after` and a conditional load `cond`), that plus the little bit of metadata definition, etc. that everyone has is all taht gets reported.

**Reply by ngocphamm** [reply]
@wbthomason Thanks for your response!

So are you saying my neovim loads in about 5ms (sequenced load + conditional load, other stuff looks super minimal)?

Because when I do `nvim --startuptime` it's reporting about 85ms.

I was looking at [this plugin](https://github.com/goolord/alpha-nvim), and it showed some profiling of plugin loading time (near the end of the link), so I thought it's from Packer, then I tried and didn't see anything similar to that.

**Reply by wbthomason** [reply]
No, the 5ms is just `packer`'s contribution. `nvim --startuptime` reports a more accurate number (although even this isn't quite the same as the "true" startup time). I would suggest trying a plugin like https://github.com/dstein64/vim-startuptime to get a global profile of your startup time - you can use `packer`'s profiling to break down the cost of `packer_compiled.lua` (which appears as a single item in profilers like the plugin I linked).

**Reply by ngocphamm** [reply]
Thank you! I will give that plugin a try!
